### PR TITLE
Docker Service Changes

### PR DESCRIPTION
**Pull request by bcongdon**
So, here are the changes to the services that need to be made for the api-gateway to work in the docker container network.

Can you think of a better way to configure this? It's going to make having a dev environment not hosted in docker difficult... Maybe have the hostname for each service live in one of the `config.go`?

### CODE REVIEW

**Comment by narendasan** [comment]
I wonder what the most ideal development flow would be, if docker is easier that non docker I think its ok. but I dont typically do dev in docker myself so I am not sure

**Comment by narendasan** [comment]
A config file may work, however we should then include in the template both versions and you could comment or uncomment what you want to use

**Comment by narendasan** [comment]
Also can you base this branch off of the arbor branch? I want to merge that first

**Comment by narendasan** [comment]
make sure to update any references to groot to groot-api-gateway (the executable should be renamed in the arbor PR)

**Comment by bcongdon** [comment]
kk, I think I hit all the necessary references

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    ln -sf /usr/src/app $GOPATH/src/github.com/acm-uiuc/groot
+
+# Create folder for client key database
+RUN mkdir -p /var/groot/
+
+# Build groot
+ADD build.sh /usr/src/app
+RUN ./build.sh
+
+CMD ["./build/groot"]
```
Comment (narendasan) [inline Dockerfile]: needs to change to the new executable

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    go install github.com/acm-uiuc/arbor/services && \
+    go install github.com/acm-uiuc/arbor/security && \
+    go install github.com/acm-uiuc/arbor/server
+
+# Bundle app source
+ADD . /usr/src/app
+
+# Create symlink to GOPATH for groot
+RUN mkdir -p $GOPATH/src/github.com/acm-uiuc && \
+    ln -sf /usr/src/app $GOPATH/src/github.com/acm-uiuc/groot
```
Comment (narendasan) [inline Dockerfile]: will need to change to acm-uiuc/groot-api-gateway

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+# Create folder for client key database
+RUN mkdir -p /var/groot/
```
Comment (narendasan) [inline Dockerfile]: /groot-api-gateway?

**Comment by narendasan** [comment]
should this be pointed at master now?

**Comment by bcongdon** [comment]
Yup

File: README.md [hunk 4]
```diff
@@ -127,6 +127,41 @@ groot [-r | --register-client client_name] [-c | --check-registration token] [-u
 > runs groot with the security layer
 
 
+## Docker
+
+#### Creating a client key:
+```
+docker-compose run groot ./build/groot -r <client_name>
```
Comment (narendasan) [inline README.md]: change to groot-api-gateway

File: README.md [hunk 5]
```diff
@@ -127,6 +127,41 @@ groot [-r | --register-client client_name] [-c | --check-registration token] [-u
...
+```
+
+#### Checking a client key: 
+```
+docker-compose run groot ./build/groot -c <client_key>
```
Comment (narendasan) [inline README.md]: here too

**Pull request merged**